app/controllers/api/v1/admin/statuses_controller.rb

```diff
@@ -28,7 +28,7 @@ def destroy
       if @status.with_media?
         # Immediately remove public copy of media instead of waiting for
         # the vacuum_orphaned_records job to take care of it later on
-        Admin::MediaAttachmentDeletionWorker.perform_inline(@status.media_attachments)
+        Admin::MediaAttachmentDeletionWorker.perform_async(@status.media_attachments.map(&:id))
```

Comment:
same change needs to be made in `status_batch_action.rb`

btw do we know how soon is the expected execution of a `perform_async` task? It should not be later than a couple minutes (if e.g. we need to take down a particularly egregious piece of media)

Reply:
Follow up for the fix!

About the timing, it's just an async call. Locally it's triggered after less than 1 second.